flink-runtime/src/test/java/org/apache/flink/runtime/metrics/groups/AbstractMetricGroupTest.java

```diff
@@ -93,6 +96,26 @@ public void testGetAllVariablesWithExclusions() {
 		assertEquals(group.getAllVariables(-1, Collections.singleton(ScopeFormat.SCOPE_HOST)).size(), 0);
 	}
 
+	@Test
+	public void testGetAllVariablesWithExclusionsForReporters() {
+		MetricRegistry registry = TestingMetricRegistry.builder().setNumberReporters(2).build();
+
+		AbstractMetricGroup<?> group = new GenericMetricGroup(registry, null, "test") {
+			@Override
+			protected void putVariables(Map<String, String> variables) {
+				variables.put("k1", "v1");
+				variables.put("k2", "v2");
+			}
+		};
+
+		group.getAllVariables(-1, Collections.emptySet());
+
+		assertThat(group.getAllVariables(0, Collections.singleton("k1")), not(IsMapContaining.hasKey("k1")));
+		assertThat(group.getAllVariables(0, Collections.singleton("k1")), (IsMapContaining.hasKey("k2")));
+		assertThat(group.getAllVariables(1, Collections.singleton("k2")), (IsMapContaining.hasKey("k1")));
```

Comment:
That's just a minor thing, but I would remove the brackets around `IsMapContaining.hasKey(..)` to make the code more readable.